tests/components/squeezebox/test_config_flow.py

```diff
@@ -0,0 +1,283 @@
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "edit"}
+    )
+
+    async def patch_async_query(self, *args):
+        self.http_status = HTTP_UNAUTHORIZED
+        return False
+
+    with patch("pysqueezebox.Server.async_query", new=patch_async_query):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOST: HOST,
+                CONF_PORT: PORT,
+                CONF_USERNAME: "test-username",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "invalid_auth"}
+
+
+async def test_form_cannot_connect(hass):
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "edit"}
+    )
+
+    async def patch_async_query(self, *args):
+        self.http_status = UNKNOWN_ERROR
+        return False
+
+    with patch("pysqueezebox.Server.async_query", new=patch_async_query):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOST: HOST,
+                CONF_PORT: PORT,
+                CONF_USERNAME: "test-username",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_discovery(hass):
+    """Test handling of discovered server."""
+    with patch(
+        "homeassistant.components.squeezebox.config_flow.SqueezeboxConfigFlow._validate_input",
+        return_value=None,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_DISCOVERY},
+            data={CONF_HOST: HOST, CONF_PORT: PORT, "uuid": UUID},
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "edit"
+
+
+async def test_discovery_no_uuid(hass):
+    """Test handling of discovered server with unavailable uuid."""
+    with patch(
+        "homeassistant.components.squeezebox.config_flow.SqueezeboxConfigFlow._validate_input",
+        return_value="invalid_auth",
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_DISCOVERY},
+            data={CONF_HOST: HOST, CONF_PORT: PORT},
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "edit"
+
+
+async def test_import(hass):
+    """Test handling of configuration imported."""
+    with patch(
+        "homeassistant.components.squeezebox.config_flow.SqueezeboxConfigFlow._validate_input",
+        return_value=None,
+    ), patch(
+        "homeassistant.components.squeezebox.async_setup", return_value=True
```

Comment:
Ok to patch `async_setup` and `async_setup_entry` as there isn't a good way to block setup otherwise